### PR TITLE
added homework_03_12

### PR DESCRIPTION
**Pull request by Viktoriyadneprovskaya**
added homework_03_12

### CODE REVIEW

File: homework_03_12_1/.gitignore [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#ignore .idea folder
```
Comment (YeaaaaH) [inline homework_03_12_1/.gitignore]: why do you add gitignore in every project?

File: homework_03_12_1/homework_03_12_1.iml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (YeaaaaH) [inline homework_03_12_1/homework_03_12_1.iml:1]: this file must be ignored

File: homework_03_12_1/src/Address.java [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+    public String getCountry() {
+        return country;
+    }
+
+    public void setCountry(String country) {
+        this.country = country;
+    }
+
+    Address(long index, String country, String region, String city, String street, int homeNumber, int flatNumber){
```
Comment (YeaaaaH) [inline homework_03_12_1/src/Address.java]: put constructor before methods

File: homework_03_12_1/src/Address.java [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+    public boolean equals(Address address){
+        if (this.index == address.index) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+    public int hashCode() {
+        return (int) index * city.hashCode() * street.hashCode() ;
```
Comment (YeaaaaH) [inline homework_03_12_1/src/Address.java]: find in internet how to do it in proper way

File: homework_03_12_1/src/Address.java [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+        address = address +"Region: "+region+"\n";
+        address = address +"City: "+city+"\n";
+        address = address +"Street: "+street+"\n";
+        address = address +"Home number: "+ homeNumber+"\n";
+        address = address +"Flat number: "+flatNumber;
+        return address;
+    }
+
+    public boolean equals(Address address){
+        if (this.index == address.index) {
```
Comment (YeaaaaH) [inline homework_03_12_1/src/Address.java]: you have 7 params and compare it only by index, find in internet how to do it in proper way

File: homework_03_12_1/src/Address.java [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+        this.homeNumber=homeNumber;
+        this.flatNumber=flatNumber;
+    }
+
+    public void setAddress(){
+
+    }
+    public String toString(){
+        String address = "";
+        address = "Index: "+ index+"\n";
```
Comment (YeaaaaH) [inline homework_03_12_1/src/Address.java]: reassign address is redundant, you can do it in one action

File: homework_03_12_1/src/Main.java [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+//создать пустой конструктор и контруктор, позволяющий заполнить поля при создании объекта
+//создать метода set,get для полей класса
+//переопределить метод equals, hashCode, toString
+
+import java.util.Scanner;
+
+public class Main {
+    public static void main(String[] args) {
+        Scanner input = new Scanner(System.in);
+        long index;
```
Comment (YeaaaaH) [inline homework_03_12_1/src/Main.java:11]: declared but not used - for all variables

File: homework_03_12_3/src/Move.java [hunk 8]
```diff
@@ -0,0 +1,5 @@
+public interface Move {
```
Comment (YeaaaaH) [inline homework_03_12_3/src/Move.java]: rename it to Moveable

File: homework_03_12_3/src/Move.java [hunk 9]
```diff
@@ -0,0 +1,5 @@
...
+    String move();
+    String voice();
```
Comment (YeaaaaH) [inline homework_03_12_3/src/Move.java]: you need to segregate interface, voice is not a part of Moveable interface

File: homework_03_12_3/src/Lion.java [hunk 10]
```diff
@@ -0,0 +1,17 @@
+public class Lion extends Cat {
+    int maxLength;
```
Comment (YeaaaaH) [inline homework_03_12_3/src/Lion.java:2]: max length of what?)

File: homework_03_12_3/src/Animal.java [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+    String habit;
+    boolean isFly;
+    boolean isMammals;
+
+    Animal(String habit,boolean isFly,boolean isMammals){
+        this.habit=habit;
+        this.isFly=isFly;
+        this.isMammals=isMammals;
+    }
+   String convert(boolean getConvert){
```
Comment (YeaaaaH) [inline homework_03_12_3/src/Animal.java:12]: getConvert is a bad example of variable name, also bad idea to create this method in Animal, replace it in Main

**Review by YeaaaaH — CHANGES_REQUESTED** [review]
need to fix comments

File: homework_03_12_3/src/Animal.java (hunk 11 above)
Comment (Viktoriyadneprovskaya) [reply homework_03_12_3/src/Animal.java:12]: I don't know how to do it in main
I got mistakes

File: homework_03_12_3/src/Lion.java (hunk 10 above)
Comment (Viktoriyadneprovskaya) [reply homework_03_12_3/src/Lion.java:2]: the lion's max body length)))

**Pull request merged**